internal/generator/vector/output/azuremonitor/azuremonitor_test.go

```diff
@@ -0,0 +1,165 @@
+package azuremonitor
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	loggingv1 "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/generator"
+	. "github.com/openshift/cluster-logging-operator/test/matchers"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	sharedKey  = "z9ndQSFH1RLDnS6WR35m84u326p3"
+	azureId    = "/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/otherResourceGroup/providers/Microsoft.Storage/storageAccounts/examplestorage"
+	hostCN     = "ods.opinsights.azure.cn"
+	customerId = "6vzw6sHc-0bba-6sHc-4b6c-8bz7sr5eggRt"
+)
+
+var _ = Describe("Generating vector config for Azure monitor output", func() {
+
+	const (
+		azureMonitorLogsDedot = `
+[transforms.azure_monitor_logs_dedot]
+type = "lua"
+inputs = ["pipelineName"]
+version = "2"
+hooks.init = "init"
+hooks.process = "process"
+source = '''
+    function init()
+        count = 0
+    end
+    function process(event, emit)
+        count = count + 1
+        event.log.openshift.sequence = count
+        if event.log.kubernetes == nil then
+            emit(event)
+            return
+        end
+        if event.log.kubernetes.labels == nil then
+            emit(event)
+            return
+        end
+		dedot(event.log.kubernetes.namespace_labels)
```

Comment:
indenting

Reply:
@jcantrill how does this fit with your plans for de-dotting, should this be somewhere more central?

Reply:
We need to review and discuss dedotting in light of it seems there are some endpoints that do not care.  I'm of the opinion we continue to dedot since that is the precedent set forth by loki and we know it is a safe way to avoid the endpoint not handling it so well, but it does seem that some are fine with it

Reply:
+1 with continue to dedot. I'm torn but there are so many broken JSON parsers out there, and we've gotten into trouble before :(  I don't think there's a strong argument to change the status quo.